### PR TITLE
Various crash fixes

### PR DESCRIPTION
**Pull request by alloy**
Probably best to look at the commits.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Artsy/Networking/ARRouter.m [hunk 1]
```diff
@@ -236,6 +236,9 @@ + (AFHTTPSessionManager *)httpClient
 
 + (void)setAuthToken:(NSString *)token
 {
+    if (token) {
```
Comment (ashfurrow) [inline Artsy/Networking/ARRouter.m]: I understand this change based on the commit message, but a comment here could prevent confusion later on.

**Comment by orta** [comment]
Same 👍 let's do it

**Comment by orta** [comment]
I think the idea of adding that comment is worth doing, so feel free to merge after rthat

File: Artsy/Networking/ARRouter.m (hunk 1 above)
Comment (alloy) [inline Artsy/Networking/ARRouter.m]: Fair 👍

**Comment by alloy** [comment]
Mergin’ hard.

**Pull request merged**